### PR TITLE
Adding unit tests to the LatticePhysicsWriter

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

I am adding unit tests to the `LatticePhysicsWriter` class.

## Why is the change being made?

This was requested by @opotowsky in ticket #1303

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/physics/neutronics/latticePhysics/tests/test_latticeWriter.py [hunk 1]
```diff
@@ -53,32 +57,130 @@ def _writeGroupStructure(self, fileObj):
...
-        """Super basic test of the LatticePhysicsWriter."""
-        o, r = loadTestReactor(TEST_ROOT)
-        cs = o.cs
-        o.cs[CONF_CROSS_SECTION].setDefaults(
-            cs[CONF_XS_BLOCK_REPRESENTATION],
-            cs[CONF_DISABLE_BLOCK_TYPE_EXCLUSION_IN_XS_GENERATION],
+    def setUp(self):
+        self.o, self.r = loadTestReactor(TEST_ROOT)
+        self.cs = self.o.cs
+        self.o.cs[CONF_CROSS_SECTION].setDefaults(
```
Comment (opotowsky) [inline armi/physics/neutronics/latticePhysics/tests/test_latticeWriter.py:63]: can this be just `self.cs`?

File: armi/physics/neutronics/latticePhysics/tests/test_latticeWriter.py [hunk 2]
```diff
@@ -53,32 +57,130 @@ def _writeGroupStructure(self, fileObj):
...
+        dfpDen["AM241"] = 3.4
+        prods = self.w._getDetailedFissionProducts(dfpDen)
+        self.assertEqual(len(prods), 3)
+        self.assertIn("U238", prods)
+        self.assertIn("U235", prods)
+        self.assertIn("AM241", prods)
+
+    def test_getDetailedFissionProductsPass(self):
+        self.cs[CONF_FP_MODEL] = "noFissionProducts"
+        self.o.cs[CONF_FP_MODEL] = "noFissionProducts"
```
Comment (opotowsky) [inline armi/physics/neutronics/latticePhysics/tests/test_latticeWriter.py:162]: related to the above question, both of these aren't needed, right? just one?

**Review by opotowsky — APPROVED** [review]
Woohoo! Thanks for doing this so quickly! Love this:

image

I'm approving since my questions are about something so tiny I don't care if it merges as-is.

**Comment by john-science** [comment]
@opotowsky I fixed your comments.

I know that doesn't _look_ like much increase in code coverage.  But this class had no unit tests before, so this PR adds real code coverage to the class and file.

**Pull request merged**